File: db4oj/core/src/com/db4o/internal/qlin/QLinField.java
/* Copyright (C) 2004 - 2010  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.qlin;

import com.db4o.qlin.*;
import com.db4o.query.*;

/**
 * @exclude
 */
public class QLinField<T> extends QLinSubNode<T>{
	
	private final Query _node;
	
	public QLinField(QLinRoot<T> root, Object expression){
		super(root);
		_node = root.descend(expression);
	}
	
	@Override
	public QLin<T> equal(Object obj) {
		Constraint constraint = _node.constrain(obj);
		constraint.equal();
		return new QLinConstraint<T>(_root, constraint);
	}
	
	@Override
	public QLin<T> startsWith(String string) {
		Constraint constraint = _node.constrain(string);
		constraint.startsWith(true);
		return new QLinConstraint<T>(_root, constraint);
	}
	
	@Override
	public QLin<T> smaller(Object obj) {
		Constraint constraint = _node.constrain(obj);
		constraint.smaller();
		return new QLinConstraint<T>(_root, constraint);
	}
	
	@Override
	public QLin<T> greater(Object obj) {
		Constraint constraint = _node.constrain(obj);
		constraint.greater();
		return new QLinConstraint<T>(_root, constraint);
	}


}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Qlin/QLinField.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Internal.Qlin;
using Db4objects.Db4o.Qlin;
using Db4objects.Db4o.Query;

namespace Db4objects.Db4o.Internal.Qlin
{
	/// <exclude></exclude>
	public class QLinField : QLinSubNode
	{
		private readonly IQuery _node;

		public QLinField(QLinRoot root, object expression) : base(root)
		{
			_node = root.Descend(expression);
		}

		public override IQLin Equal(object obj)
		{
			IConstraint constraint = _node.Constrain(obj);
			constraint.Equal();
			return new QLinConstraint(((QLinRoot)_root), constraint);
		}

		public override IQLin StartsWith(string @string)
		{
			IConstraint constraint = _node.Constrain(@string);
			constraint.StartsWith(true);
			return new QLinConstraint(((QLinRoot)_root), constraint);
		}

		public override IQLin Smaller(object obj)
		{
			IConstraint constraint = _node.Constrain(obj);
			constraint.Smaller();
			return new QLinConstraint(((QLinRoot)_root), constraint);
		}

		public override IQLin Greater(object obj)
		{
			IConstraint constraint = _node.Constrain(obj);
			constraint.Greater();
			return new QLinConstraint(((QLinRoot)_root), constraint);
		}
	}
}
